SD_Assignment1/src/main/java/ro/utcn/aron/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,84 @@
+package ro.utcn.aron.jdbc;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Component;
+
+import ro.utcn.aron.api.QuestionRepository;
+import ro.utcn.aron.model.Question;
+
+@Component
+public class JdbcQuestionRepository implements QuestionRepository {
+
+	private  JdbcTemplate template; //final
+	
+	@Override
+	public Question save(Question question) {
+		if(question.getId() != 0) {
+			update(question);
+		} else {
+			int id = insert(question);
+			question.setId(id);
+		}
+		return null;
+	}
+
+	@Override
+	public Optional<Question> findByID(int id) {
+		List<Question> questions = template.query("SELECT * FROM question WHERE id = ?"+id, 

```

Comment:
Why are you concatenating the ID into the query? You are already using prepared statement parameters.